mycore-fo/src/main/java/org/mycore/component/fo/common/content/xml/MCRLayoutTransformerFoFactory.java

```diff
@@ -0,0 +1,160 @@
+
+    private static MCRFopper fopper = new MCRFopper();
+
+    private static final MCRIdentityTransformer NOOP_TRANSFORMER = new MCRIdentityTransformer("text/xml", "xml");
+
+    /**
+     * Returns the transformer with the given ID. If the transformer is not instantiated yet,
+     * it is created and initialized.
+     */
+    @Override
+    public MCRContentTransformer getTransformer(String id) throws Exception {
+        MCRContentTransformer transformer = transformers.get(id);
+        if (transformer != null) {
+            return transformer;
+        }
+        //try to get configured transformer
+        transformer = MCRContentTransformerFactory.getTransformer(id.replaceAll("-default$", ""));
+        if (transformer != null) {
+            transformers.put(id, transformer);
+            return transformer;
+        }
+        return buildLayoutTransformer(id);
+    }
+
+    private static MCRContentTransformer buildLayoutTransformer(String id) throws Exception {
+        String idStripped = id.replaceAll("-default$", "");
+        LOGGER.debug("Configure property MCR.ContentTransformer.{}.Class if you do not want to use default behaviour.",
+            idStripped);
+        String stylesheet = getResourceName(id);
+        if (stylesheet == null) {
+            LOGGER.debug("Using noop transformer for {}", idStripped);
+            return NOOP_TRANSFORMER;
+        }
+        String[] stylesheets = getStylesheets(idStripped, stylesheet);
+        MCRContentTransformer transformer = MCRXSLTransformer.getInstance(stylesheets);
+        String mimeType = transformer.getMimeType();
+        if (isPDF(mimeType)) {
+            transformer = new MCRTransformerPipe(transformer, fopper);
+            LOGGER.debug("Using stylesheet '{}' for {} and MCRFopper for PDF output.", Lists.newArrayList(stylesheets),
+                idStripped);
+        } else {
+            LOGGER.debug("Using stylesheet '{}' for {}", Lists.newArrayList(stylesheets), idStripped);
+        }
+        transformers.put(id, transformer);
+        return transformer;
+    }
+
+    private static boolean isPDF(String mimeType) {
+        return MediaType.parse(mimeType).is(MediaType.PDF);
+    }
+
+    private static String[] getStylesheets(String id, String stylesheet)
+        throws TransformerException, SAXException, ParserConfigurationException {
+        List<String> ignore = MCRConfiguration.instance().getStrings("MCR.LayoutTransformerFactory.Default.Ignore",
+            Collections.emptyList());
+        List<String> defaults = Collections.emptyList();
+        if (!ignore.contains(id)) {
+            MCRXSLTransformer transformerTest = MCRXSLTransformer.getInstance(stylesheet);
+            String outputMethod = transformerTest.getOutputProperties().getProperty(OutputKeys.METHOD, "xml");
+            if ("xml".equals(outputMethod) && !isPDF(transformerTest.getMimeType())) {
```

Comment:
You can put a method named (isXMLOutput)to the DefaultTransformerFactory which just does the "xml".equals(outputMethod) part and then overwrite it in the MCRLayoutTransformerFOFactory and add super.isXMLOutput() && && !isPDF(transformerTest.getMimeType()) .